File: Client/src/app/core/models/signal-r/update-note-ws.ts
import { Label } from 'src/app/content/labels/models/label.model';
import { MergeTransaction } from 'src/app/content/notes/full-note/content-editor/text/rga/types';

export interface UpdateNoteWS {
  noteId: string;
  color: string;
  titleTransaction: MergeTransaction<string>;
  isUpdateTitle: boolean;
  removeLabelIds: string[];
  addLabels: Label[];
}

File: Backend/NootsWEB/Common/DTO/WebSockets/UpdateNoteWS.cs
using Common.DTO.Labels;
using Noots.RGA_CRDT;
using System;
using System.Collections.Generic;

namespace Common.DTO.WebSockets
{
    public class UpdateNoteWS
    {
        public Guid NoteId { set; get; }

        public string Color { set; get; }

        public MergeTransaction<string> TitleTransaction { set; get; }

        public bool IsUpdateTitle { set; get; }

        public List<Guid> RemoveLabelIds { set; get; }

        public List<LabelDTO> AddLabels { set; get; }


    }
}
